service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -195,6 +189,50 @@ public ControlledGcpResourceApiController(
         workspaceId, resourceId, userRequest, r -> r.castToGcsBucketResource().toApiResource());
   }
 
+  @Override
+  public ResponseEntity<ApiCloneControlledGcpGcsBucketResult> cloneGcsBucket(
+      UUID workspaceId, UUID resourceId, @Valid ApiCloneControlledGcpGcsBucketRequest body) {
+    logger.info("Cloning GCS bucket resourceId {} workspaceId {}", resourceId, workspaceId);
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    final ControlledResource sourceResource =
+        controlledResourceService.getControlledResource(workspaceId, resourceId, userRequest);
+    final ControlledGcsBucketResource sourceBucket = sourceResource.castToGcsBucketResource();
+    final String jobId =
+        controlledResourceService.cloneGcsBucket(
+            sourceBucket,
+            body.getDestinationWorkspaceId(),
+            body.getCloningInstructions(),
+            body.getJobControl(),
+            body.getLocation(),
+            body.getName(),
+            body.getDescription(),
+            body.getBucketName(),
+            userRequest);
+    final ApiCloneControlledGcpGcsBucketResult result =
+        fetchCloneGcsBucketResult(jobId, userRequest);
+    return new ResponseEntity<>(result, HttpStatus.valueOf(result.getJobReport().getStatusCode()));
+  }
+
+  private ApiCloneControlledGcpGcsBucketResult fetchCloneGcsBucketResult(
+      String jobId, AuthenticatedUserRequest userRequest) {
+    final AsyncJobResult<ApiClonedControlledGcpGcsBucket> jobResult =
+        jobService.retrieveAsyncJobResult(
+            jobId, ApiClonedControlledGcpGcsBucket.class, userRequest);
+    // TODO: generate an intermediate class if necessary, beneath the API layer
+    return new ApiCloneControlledGcpGcsBucketResult()
+        .jobReport(jobResult.getJobReport())
+        .errorReport(jobResult.getApiErrorReport())
+        .bucket(jobResult.getResult());
+  }
+
+  @Override
+  public ResponseEntity<ApiCloneControlledGcpGcsBucketResult> getCloneGcsBucketResult(
+      UUID workspaceId, String jobId) {
```

Comment:
Never mind. I think we have a more general access control problem, but outside the scope of this PR.

Our JobService will make sure that the subjectId of the submitter is the same as the subjectId on the `retrieveAsyncJobResult`.  That is not quite enough for access control. It means that someone who has been kicked out of the Workspace or even out of the company could still hit the interface and get job results. I made PF-859 to capture the problem.

Reply:
Just FYI - within the JobService, we have access to the input parameters of the flight as well as the results, so it would be possible to pull the initial request from there and find the destination workspace.